metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -310,6 +311,35 @@ def _register_job_definition(
                 }
             ]
 
+        if efa:
+            if not (isinstance(efa, (int, unicode, basestring))):
+                raise BatchJobException(
+                    "Invalid efa value: ({}) (should be 0 or greater)".format(
+                        efa
+                    )
+                )
+            else:
+                job_definition["containerProperties"]["linuxParameters"]["devices"] = []
```

Comment:
@rileyhun @emattia can you add a comment here on why there is a different treatment for num_parallel for future readers.